src/konfetti/vault/asynchronous.py

```diff
@@ -32,20 +32,23 @@ def cached_call(method):
 class AsyncVaultBackend(BaseVaultBackend):
     is_async = True
 
-    async def load(self, path, url, token):
+    async def load(self, path, url, token, username, password):  # pylint: disable=too-many-arguments
         from aiohttp import ClientConnectionError
         from tenacity import AsyncRetrying
 
         retry = self._get_retry(AsyncRetrying, ClientConnectionError)
         full_path = self._get_full_path(path)
-        return await retry.call(self._call, full_path, url, token)
+        return await retry.call(self._call, full_path, url, token, username, password)
 
     @cached_call
-    async def _call(self, path, url, token):
+    async def _call(self, path, url, token, username, password):  # pylint: disable=too-many-arguments
         """A call to the Vault server."""
         vault_logger.debug('Access "%s" in Vault', path)
         import aiohttp
 
+        if username and password:
```

Comment:
What if the token is here as well? should it be preferred instead?

Reply:
Maybe we need to try the token and if the call failed, then retry with username & password?
Another idea - cache the new token somehow and reuse it?